test/common/init/target_impl_test.cc

```diff
@@ -60,6 +62,110 @@ TEST(InitTargetImplTest, ReadyWhenWatcherUnavailable) {
   EXPECT_FALSE(target.ready());
 }
 
+TEST(SharedTargetImplTest, OnSyncManagers) {
+  ManagerImpl m1("m1");
+  ManagerImpl m2("m2");
+  int counter = 0;
+  SharedTargetImpl t("test", [&counter]() mutable { counter++; });
+  m1.add(t);
+  EXPECT_EQ(0, counter);
+  m2.add(t);
+  EXPECT_EQ(0, counter);
+  EXPECT_EQ(Manager::State::Uninitialized, m1.state());
+  EXPECT_EQ(Manager::State::Uninitialized, m2.state());
```

Comment:
Please add comments accompanying all these `EXPECT` statements to help explain the logic of the test; see style of existing tests.